cmd/mt-whisper-importer-reader/main.go

```diff
@@ -252,48 +252,47 @@ func sortPoints(points pointSorter) pointSorter {
 	return points
 }
 
-func shortAggMethodString(aggMethod whisper.AggregationMethod) (string, error) {
-	switch aggMethod {
+func convertWhisperMethod(whisperMethod whisper.AggregationMethod) (schema.Method, error) {
+	switch whisperMethod {
 	case whisper.AggregationAverage:
-		return "avg", nil
+		return schema.Avg, nil
 	case whisper.AggregationSum:
-		return "sum", nil
-	case whisper.AggregationMin:
-		return "min", nil
-	case whisper.AggregationMax:
-		return "max", nil
+		return schema.Sum, nil
 	case whisper.AggregationLast:
-		return "lst", nil
+		return schema.Lst, nil
+	case whisper.AggregationMax:
+		return schema.Max, nil
+	case whisper.AggregationMin:
+		return schema.Min, nil
 	default:
-		return "", fmt.Errorf("Unknown aggregation method %d", aggMethod)
+		return 0, fmt.Errorf("Unknown whisper method: %d", whisperMethod)
 	}
 }
 
-func getMetric(w *whisper.Whisper, file, name string) (archive.Metric, error) {
-	res := archive.Metric{
-		AggregationMethod: uint32(w.Header.Metadata.AggregationMethod),
+func getMetric(w *whisper.Whisper, file, name string) (mdata.ArchiveRequest, error) {
+	res := mdata.ArchiveRequest{
```

Comment:
we should move this to after checking that the whisper file has data.  No point in allocating memory if we dont need to.